Notifications/NotificationService.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  NotificationService.swift
+//  Notifications
+//
+//  Created by Ezequiel Becerra on 24/01/2018.
+//  Copyright © 2018 Ezequiel Becerra. All rights reserved.
+//
+
+import UserNotifications
+
+class NotificationService: UNNotificationServiceExtension {
+    
+    var contentHandler: ((UNNotificationContent) -> Void)?
+    var bestAttemptContent: UNMutableNotificationContent?
+    
+    override func didReceive(_ request: UNNotificationRequest,
+                             withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        
+        func save(_ identifier: String, data: Data, options: [AnyHashable: Any]?) -> UNNotificationAttachment? {
+            
+            var directory = URL(fileURLWithPath: NSTemporaryDirectory())
+            directory = directory.appendingPathComponent(
+                ProcessInfo.processInfo.globallyUniqueString,
+                isDirectory: true)
+            
+            do {
+                try FileManager.default.createDirectory(at: directory, withIntermediateDirectories: true, attributes: nil)
+                
+                let fileURL = directory.appendingPathComponent(identifier)
+                try data.write(to: fileURL, options: [])
+                return try UNNotificationAttachment.init(identifier: identifier, url: fileURL, options: options)
+            } catch {
+                //  Do nothing
+            }
+            return nil
+        }
+        
+        func exitGracefully(_ reason: String = "") {
+            let bca = request.content.mutableCopy() as? UNMutableNotificationContent
+            bca!.title = reason
+            contentHandler(bca!)
+        }
+        
+        guard let content = (request.content.mutableCopy() as? UNMutableNotificationContent) else {
+            return exitGracefully()
+        }
+        
+        let userInfo: [AnyHashable: Any] = request.content.userInfo
+        
+        guard let attachmentURL = userInfo["media-url"] as? String else {
+            return exitGracefully()
+        }
+        
+        guard let imageData  =
+            try? Data(contentsOf: URL(string: attachmentURL)!)
+            else {
+                return exitGracefully()
+        }
+        
+        guard let attachment =
+            save("image.png", data: imageData, options: nil)
+            else {
+                return exitGracefully()
+        }
+        
+        content.attachments = [attachment]
+        contentHandler(content.copy() as! UNNotificationContent)
+    }
+    
+    override func serviceExtensionTimeWillExpire() {
+        // Called just before the extension will be terminated by the system.
+        // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/2116f8d687ff9cf2ddfe00a13535cdb270d1c54c/Notifications/NotificationService.swift#L72"></span>Line should be 120 characters or less: currently 139 characters</td>
    </tr>
  </tbody>
</table>